app/src/main/java/com/sjh/sunflower_sksowk156/MainActivity.kt

```diff
@@ -1,46 +1,177 @@
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting("Android")
-                }
+                MainScreen()
             }
         }
     }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Text(
-        text = "Hello $name!",
-        modifier = modifier
-    )
+fun MainScreen() {
+    val tabs = listOf("My garden", "Plant list")
+    val pagerState = rememberPagerState(pageCount = { tabs.size })
+    val showTopBar = remember { mutableStateOf(true) }
+
+    // NestedScrollConnection을 사용하여 스크롤에 따라 TopAppBar를 숨기거나 표시
+    val nestedScrollConnection = remember {
+        object : NestedScrollConnection {
+            override fun onPreScroll(available: Offset, source: NestedScrollSource): Offset {
+                // 스크롤 방향에 따라 TopAppBar의 보이기/숨기기 상태를 변경
+                val delta = available.y
+                showTopBar.value = delta > 0 || (delta == 0f && showTopBar.value)
+                return Offset.Zero
+            }
+        }
+    }
+    Scaffold(
+        modifier = Modifier.nestedScroll(nestedScrollConnection)
+    ) { paddingValues ->
+        Column(modifier = Modifier.fillMaxSize()) {
+            MainToolbar(showTopBar = showTopBar, currentPage = pagerState.currentPage)
+            MainContent(pagerState = pagerState, tabs = tabs)
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun MainContent(tabs: List<String>, pagerState: PagerState) {
+    val coroutineScope = rememberCoroutineScope()
+    TabRow(
+        selectedTabIndex = pagerState.currentPage, modifier = Modifier.fillMaxWidth()
+    ) {
+        tabs.forEachIndexed { index, title ->
+            Tab(text = { Text(text = title) },
+                selected = pagerState.currentPage == index,
+                onClick = {
+                    coroutineScope.launch {
+                        pagerState.animateScrollToPage(index)
+                    }
+                })
+        }
+    }
+
+    HorizontalPager(
+        state = pagerState, modifier = Modifier.fillMaxHeight()
+    ) { page ->
+        when (page) {
+            0 -> MyGardenScreen()
+            1 -> PlantList()
+        }
+    }
+}
+
+@Composable
+fun MainToolbar(showTopBar: MutableState<Boolean>, currentPage: Int) {
+    AnimatedVisibility(
+        visible = showTopBar.value, enter = slideInVertically(
+            // TopAppBar가 아래로 나타날 때
+            initialOffsetY = { -it },
+            animationSpec = tween(durationMillis = 300, easing = LinearOutSlowInEasing)
+        ), exit = slideOutVertically(
+            // TopAppBar가 위로 사라질 때
+            targetOffsetY = { -it },
+            animationSpec = tween(durationMillis = 300, easing = LinearOutSlowInEasing)
+        )
```

Comment:
재미있는 기능을 구현하고 계시네요.
테스트를 해보았습니다만, 애니메이션이 끝날 때 갑자기 visibility가 전환되어서, mis-tapping을 유발할 것 같은 느낌이 있습니다. 그 부분의 사용성을 개선하면 좋을 것 같네요.
```suggestion
        visible = showTopBar.value,
        enter = slideInVertically(
            // TopAppBar가 아래로 나타날 때
            initialOffsetY = { -it },
            animationSpec = tween(durationMillis = 300, easing = LinearOutSlowInEasing)
        ),
        exit = slideOutVertically(
            // TopAppBar가 위로 사라질 때
            targetOffsetY = { -it },
            animationSpec = tween(durationMillis = 300, easing = LinearOutSlowInEasing)
        )
```